test/dsl/dsl_options.jl

```diff
@@ -1128,55 +1145,184 @@ let
+        @differentials D = Differential(t)
+    end
+    rn42 = @reaction_network rn1 begin
+        @differentials begin
+            D = Differential(t)
+        end
+    end
+    @test isequal(rn41, rn42)
+    @test_throws Exception @eval @reaction_network begin
+        @differentials D = Differential(t) Δ = Differential(t)
+    end
+
+    # The `@continuous_events` option.
+    rn51 = @reaction_network rn1 begin
+        @species X(t)
+        @continuous_events [X ~ 3.0] => [X ~ X - 1]
+    end
+    rn52 = @reaction_network rn1 begin
+        @species X(t)
+        @continuous_events begin
+            [X ~ 3.0] => [X ~ X - 1]
+        end
+    end
+    @test isequal(rn51, rn52)
+    @test_throws Exception @eval @reaction_network begin
+        @species X(t)
+        @continuous_events [X ~ 3.0] => [X ~ X - 1] [X ~ 1.0] => [X ~ X + 1]
+    end
+
+    # The `@discrete_events` option.
+    rn61 = @reaction_network rn1 begin
+        @species X(t)
+        @discrete_events [X > 3.0] => [X ~ X - 1]
+    end
+    rn62 = @reaction_network rn1 begin
+        @species X(t)
+        @discrete_events begin
+            [X > 3.0] => [X ~ X - 1]
+        end
+    end
+    @test isequal(rn61, rn62)
+    @test_throws Exception @eval @reaction_network begin
+        @species X(t)
+        @discrete_events [X > 3.0] => [X ~ X - 1] [X < 1.0] => [X ~ X + 1]
+    end
 end
 
 # test combinatoric_ratelaws DSL option
 let
+    # Test for `@combinatoric_ratelaws false`.
     rn = @reaction_network begin
         @combinatoric_ratelaws false
         (k1,k2), 2A <--> B
-        end
+    end
     combinatoric_ratelaw = Catalyst.get_combinatoric_ratelaws(rn)
     @test combinatoric_ratelaw == false
     rl = oderatelaw(reactions(rn)[1]; combinatoric_ratelaw)
     @unpack k1, A = rn
     @test isequal(rl, k1*A^2)
 
+    # Test for `@combinatoric_ratelaws true`.
     rn2 = @reaction_network begin
         @combinatoric_ratelaws true
         (k1,k2), 2A <--> B
-        end
+    end
     combinatoric_ratelaw = Catalyst.get_combinatoric_ratelaws(rn2)
     @test combinatoric_ratelaw == true
     rl = oderatelaw(reactions(rn2)[1]; combinatoric_ratelaw)
     @unpack k1, A = rn2
     @test isequal(rl, k1*A^2/2)
 
+    # Test for interpolation into `@combinatoric_ratelaws`.
     crl = false
     rn3 = @reaction_network begin
         @combinatoric_ratelaws $crl
         (k1,k2), 2A <--> B
-        end
+    end
     combinatoric_ratelaw = Catalyst.get_combinatoric_ratelaws(rn3)
     @test combinatoric_ratelaw == crl
     rl = oderatelaw(reactions(rn3)[1]; combinatoric_ratelaw)
     @unpack k1, A = rn3
     @test isequal(rl, k1*A^2)
+
+    # Test for erroneous inputs (to few, to many, wrong type).
+    @test_throws Exception @eval @reaction_network begin
+        @combinatoric_ratelaws
+        d, 3X --> 0
+    end
+    @test_throws Exception @eval @reaction_network begin
+        @combinatoric_ratelaws false false
+        d, 3X --> 0
+    end
```

Comment:
Checks that new error checks work (don't think the one above actually errored either, but it should)